### PR TITLE
feat: increase LimitNOFILE for both celestia-app and celestia-node

### LINKED ISSUE #1390: `LimitNOFILE` env var in systemd should be higher
This is the page they should update: https://docs.celestia.org/nodes/systemd

What Happens?

- badgerdb uses extensive files to run 
    - Badger keeps the file descriptors open for SST and vlog files.
    - badger doesn't release the file descriptors until the db (or the file) is closed. 
- Celestia Docs for systemd set the ulimit at 4096 LimitNOTFILE=4096
- Badger themselves say it needs a lot of open files to run properly in production (>1million)
    - https://dgraph.io/docs/deploy/troubleshooting/#too-many-open-files-errors
- Even if you set DefaultLimitNoFILE to a large number for ALL processes, by setting the value in systemd it will use the systemd limit

From our bridge node, current open files in badgerDB - `1330736`

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

Resolves #1390. [An issue was been reported cites the minimum required by badger is 1million+](https://github.com/celestiaorg/docs/issues/1390) and their bridge node requires 1330736 files open in badgerDB, but we currently have documented 4096, 65536, 65536 for light/full/bridge and 4096 documented for celestia-app. 

This PR updates the values to ones suggested in the issue or in slack.

| software | previous value | proposed value |
| -------- | ------------- | ----------------|
| celestia-app | 4096 | 65535 |
| celestia-node light | 4096 | 1400000 |
| celestia-node full | 65536 | 1400000 |
| celestia-node bridge | 65536 | 1400000 |

This needs review from someone on these teams, as it impacts suggested parameter changes relevant to all of the teams:
- [x] @celestiaorg/celestia-node 
- [x] @celestiaorg/celestia-core 
- [ ] @celestiaorg/devops 

## Reference

From [badgerdb's documentation](https://dgraph.io/docs/deploy/troubleshooting/#too-many-open-files-errors): 
image

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

## Summary by CodeRabbit

- **Documentation**
	- Updated the systemd configuration guide to reflect increased file limit settings for service units, enhancing system performance and reliability.

### CODE REVIEW

File: nodes/systemd.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ User=$USER
 ExecStart=$(which celestia-appd) start
 Restart=on-failure
 RestartSec=3
-LimitNOFILE=4096
+LimitNOFILE=16384
```
[the remark this answers is not in the document]
Comment (jcstein) [reply nodes/systemd.md:33]: Using @rootulp's last suggestion from Slack.

**Review by renaynay — APPROVED** [review]

**Comment by renaynay** [comment]
Celestia light node shouldn't need that high of a limit though

**Comment by jcstein** [comment]
Does it use badgerdb? Badegrdb suggests "a soft limit of 1048576"

**Comment by Wondertan** [comment]
I think we should remove this configuration from LN. The default hard limit in systemd is high enough(524288) for LN and since go1.20, the [runtime](https://go-review.googlesource.com/c/go/+/392415) gets up to the hard limit automatically.

**Comment by Wondertan** [comment]
@jcstein, LN uses badgerdb, but note that the shared link about limits is from Dgraph database which uses badgerdb underneath and it's for the badger itself. Badger needs less then that by default. 

In case of FN/BN where we have multiple instances of badger and CAR files IO we definitely need more than badger suggests and the proposed value makes total sense.

**Comment by jcstein** [comment]
Screenshot 2024-02-12 at 6 20 21 PM

Screenshot 2024-02-12 at 6 20 50 PM

i'm not a db expert, but i'm nearly positive this is badgerDBs official documentation? and the google results, for example, only show other things which don't mention "db"

**Comment by jcstein** [comment]
> I think we should remove this configuration from LN. The default hard limit in systemd is high enough(524288) for LN and since go1.20, the [runtime](https://go-review.googlesource.com/c/go/+/392415) gets up to the hard limit automatically.

I'll suggest this on the pr

File: nodes/systemd.md [hunk 2]
```diff
@@ -182,7 +182,7 @@ User=$USER
 ExecStart=$(which celestia) light start --core.ip <URI>
 Restart=on-failure
 RestartSec=3
-LimitNOFILE=4096
+LimitNOFILE=1400000
```
[the remark this answers is not in the document]
Comment (jcstein) [reply nodes/systemd.md:185]: cc @celestiaorg/celestia-node

File: nodes/systemd.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ User=$USER
 ExecStart=$(which celestia-appd) start
 Restart=on-failure
 RestartSec=3
-LimitNOFILE=4096
+LimitNOFILE=65535
```
Comment (rootulp) [inline nodes/systemd.md:33]: 👍 LGTM b/c this is what POPS uses.

**Review by rootulp — APPROVED** [review]
Thanks for following up on this @jcstein 👍

**Pull request merged**